prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -1446,12 +1446,18 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
                 let (restored, r_stmts, ..) = self.encode_place(place, ArrayAccessKind::Shared)?;
                 stmts.extend(r_stmts);
 
-                (expiring_base, restored, false, stmts)
+                (expiring_base, Some(restored), false, stmts)
             }
 
             mir::Rvalue::Use(mir::Operand::Constant(box mir::Constant { literal, .. })) => {
                 let (ty, val) = mir_constantkind_to_ty_val(literal);
-                let restored = self.encoder.encode_const_expr(ty, &val)?;
+
+                let restored : Option<vir::Expr> = match ty.kind() {
+                    ty::TyKind::Ref(_, inner, _) if inner.is_str() => None,
```

Comment:
I think you need to check if the reference is `'static`. Otherwise, this is probably unsound.

Reply:
It looks like this is already in the case branch for a literal. If it's literal, shouldn't it be `'static` in principle?

In fact it's not in practice, I see that sometimes the region is `'_#3r` instead of `ReStatic`, which I expected.

But anyways I think the check here needs only to be checking for a string. Regardless of the region of the string literal, currently we dont encode a VIR constant expression for it.

Reply:
Yes, I think you are right. Could you please add a comment here and below explaining that?